core/src/main/java/feast/core/job/JobUpdateTask.java

```diff
@@ -73,147 +74,131 @@ public JobUpdateTask(
     this.currentJob = currentJob;
     this.jobManager = jobManager;
     this.jobUpdateTimeoutSeconds = jobUpdateTimeoutSeconds;
+    this.runnerType = jobManager.getRunnerType().toString();
   }
 
   @Override
   public Job call() {
     ExecutorService executorService = Executors.newSingleThreadExecutor();
-    Source source = Source.fromProto(sourceSpec);
     Future<Job> submittedJob;
-    if (currentJob.isPresent()) {
-      Set<String> existingFeatureSetsPopulatedByJob =
-          currentJob.get().getFeatureSets().stream()
-              .map(FeatureSet::getId)
-              .collect(Collectors.toSet());
-      Set<String> newFeatureSetsPopulatedByJob =
-          featureSets.stream()
-              .map(fs -> FeatureSet.fromProto(fs).getId())
-              .collect(Collectors.toSet());
-      if (existingFeatureSetsPopulatedByJob.size() == newFeatureSetsPopulatedByJob.size()
-          && existingFeatureSetsPopulatedByJob.containsAll(newFeatureSetsPopulatedByJob)) {
```

Comment:
This was [the definition of `Set#equals`][1] ☺️ 

[1]: https://github.com/openjdk/jdk/blob/5fc5cb9f252310634e36ed2ce2e09fa42e430bcf/src/java.base/share/classes/java/util/AbstractSet.java#L85-L99